docs/client-sdk/javascript/tutorials/use-a-persistent-conversation-cache.md

```diff
@@ -0,0 +1,155 @@
+
+```tsx
+import { create } from "zustand";
+import { persist } from "zustand/middleware";
+import { ConversationExport } from "@xmtp/xmtp-js/dist/types/src/conversations/Conversation";
+import { getEnv } from "../helpers";
+
+export const CONVERSATION_CACHE_VERSION = 1;
+
+/**
+ * The ConversationCache is a JSON serializable Zustand store that is persisted to LocalStorage
+ * Persisting conversations to the cache saves on both bandwidth and CPU cycles, as we don't have to re-fetch or re-decrypt conversations on subsequent page loads
+ */
+interface ConversationCache {
+  // Mapping of conversation exports, keyed by wallet address
+  conversations: { [walletAddress: string]: ConversationExport[] };
+  // Overwrite the cache for a given wallet address
+  setConversations: (
+    walletAddress: string,
+    conversations: ConversationExport[],
+  ) => void;
+  // Add a single conversation to the cache.
+  // Deduping only happens at the time the cache is loaded, so be careful to not overfill or you will use more LocalStorage space than necessary
+  addConversation: (
+    walletAddress: string,
+    conversation: ConversationExport,
+  ) => void;
+}
+
+export const useConversationCache = create<ConversationCache>()(
+  persist(
+    (set, get) => ({
+      conversations: {},
+      setConversations: (walletAddress: string, convos: ConversationExport[]) =>
+        set({
+          conversations: { ...get().conversations, [walletAddress]: convos },
+        }),
+      addConversation: (walletAddress: string, convo: ConversationExport) => {
+        const existing = get().conversations;
+        const existingForWallet = existing[walletAddress] || [];
+        return set({
+          conversations: {
+            ...existing,
+            [walletAddress]: [...existingForWallet, convo],
+          },
+        });
+      },
+    }),
+    {
+      // Ensure that the LocalStorage key includes the network and the cache version
+      // If any breaking changes to the ConversationExport schema occur, increment the cache version.
+      name: `xmtp:conversations:${getEnv()}:v${CONVERSATION_CACHE_VERSION}`,
+      partialize: (state) => ({ conversations: state.conversations }),
+    },
+  ),
+);
+```
+
+For the `getEnv` import, see [`helpers/index.ts`](https://github.com/xmtp-labs/xmtp-inbox-web/blob/main/helpers/index.ts) and [`helpers/env.ts`](https://github.com/xmtp-labs/xmtp-inbox-web/blob/main/helpers/env.ts) in the example XMTP Inbox web app repo.
+
+## Add conversations to the persistent cache
+
+Add existing and new conversations to the persistent cache.
+
+For example, see these code snippets from [`hooks/useListConversations.tsx`](https://github.com/xmtp-labs/xmtp-inbox-web/blob/main/hooks/useListConversations.tsx) in the example XMTP Inbox web app:
+
+```tsx
+import { useConversationCache } from "../store/conversationCache";
+
+const setConversationCache = useConversationCache(
+    (state) => state.setConversations,
+  );
+
```

Comment:
Thank you, @neekolas:
- Removed the `const addToConversationCache = useConversationCache( ...` function.
- Replaced all of the pre-existing code below with the `useEffect` you provided.